Model/Indexer/Product/Indexer.php

```diff
@@ -57,14 +57,21 @@ public function __construct(
     public function executeFull()
     {
         if ($this->dataHelper->isFullReindexEnabled()) {
-            // Fetch all enabled products
-            $searchCriteria = $this->searchCriteriaBuilder
-                ->addFilter('status', Status::STATUS_ENABLED, 'eq')
-                ->setPageSize(self::HARD_LIMIT_FOR_PRODUCTS)
-                ->setCurrentPage(1)
-                ->create();
-            $products = $this->productRepository->getList($searchCriteria);
-            $this->productService->update($products->getItems());
+            $pageNumber = 0;
+            $totalItems = 0;
+            do {
+                $pageNumber++;
+
+                $searchCriteria = $this->searchCriteriaBuilder
+                    ->addFilter('status', Status::STATUS_ENABLED, 'eq')
+                    ->setPageSize(self::BATCH_SIZE)
+                    ->setCurrentPage($pageNumber)
+                    ->create();
+                $products = $this->productRepository->getList($searchCriteria);
```

Comment:
check the return value before sending it to the update method

Reply:
Was it being checked before?

Reply:
no, but it could return empty now. For say the $totalItems % 1000 == 0, it could case issue.